chat/conversations.ts

```diff
@@ -33,6 +33,28 @@ import isChannel = Interfaces.isChannel;
  */
 const CONVERSATION_CACHE_UPDATE_FREQ_IN_MS = 1000;
 
+
+function shouldEnableLogging(
+  conversation: Interfaces.Conversation,
+  isAd: boolean = false
+): boolean {
+  const loggingSetting = conversation.settings.enableLogging;
+  
+  // If explicitly set to True or False, use that
+  if (loggingSetting === Interfaces.Setting.True) {
+    return true;
+  }
+  if (loggingSetting === Interfaces.Setting.False) {
+    return false;
+  }
+  
+  // Otherwise (Default), use global settings
+  if (isAd) {
```

Comment:
Will this check ever be hit if you have the setting set to a non-default value? Seems to me like it will now only check for the ad setting if you have the conversation-level setting set to default, which I doubt is intended.

I'll add that I haven't tested that specific test case myself, but did you?